File: data-structures/arrays/dynamic/java/dynamicArray.java

import java.util.ArrayList;
import java.util.random.RandomGenerator;

public class dynamicArray {

    public static void main(String[] args) {
        /* 
         * Dynamic Arrays in Java
         * These can be implemented using ArrayList or similar data structures.
         * Here is a simple example using ArrayList:
         */

         ArrayList<Integer> arr = new ArrayList<>();

         for (int i = 0; i < 10; i++) {
            arr.add(RandomGenerator.getDefault().nextInt(1, 10));   // adding random integers to the array
         }

         System.out.println("""
                 dynamic array 'arr' elements:
                 ----------------------------
                 """);
         for (int i = 0; i < arr.size(); i++) {
            System.out.println("element at index " + (i + 1) + ": \t" + arr.get(i)); // accessing elements
         }
    }       
}

File: data-structures/arrays/dynamic/python/dynamicArray.py
import random

""" 
DYNAMIC ARRAYS
---------------

this is an example of a dynamic array in python using lists.
"""

arr = []    # creating an empty dynamic array

for i in range(10):
    arr.append(random.randint(1, 10))  # appending elements to the dynamic array


# outputing elements to the standard output
print("dynamic array 'arr' elements:\n------------------------------------")
for i in range(len(arr)):
    print("element", (i + 1), "\t", arr[i])  # printing the elements of the dynamic array
